src/core/public/application/types.ts

```diff
@@ -75,7 +75,7 @@ export interface AppBase {
 export interface App extends AppBase {
   /**
    * A mount function called when the user navigates to this app's route.
-   * @param context The mount context for this app.
+   * @param context The mount context for this app. Deprecated, use {@link CoreSetup.getStartServices}.
    * @param targetDomElement An HTMLElement to mount the application onto.
    * @returns An unmounting function that will be called to unmount the application.
    */
```

Comment:
Should we expose a non-deprecated version of `mount` on the interface instead of only a comment?

```
/** @deprecated */
mount: (context: AppMountContext, params: AppMountParameters) => AppUnmount | Promise<AppUnmount>;
mount: (params: AppMountParameters) => AppUnmount | Promise<AppUnmount>;
```

That forces the implementation to typecheck the parameters (or the length of them), but that may be better?

Reply:
I like this idea, I'll add that.